release notes/v0.40.0.md

```diff
@@ -0,0 +1,285 @@
+
+Important to highlight that the **k6 team does not guarantee backward compatibility for these modules** and may change or remove them altogether. Also, their import paths, starting with `k6/experimental`, will break when the modules stop being experimental. Of course, we are going to try to limit those breaking changes to a minimum and, when possible, do them in a backward compatible way for at least a version.
+
+<details>
+<summary> Redis example </summary>
+
+Here is a fairly big example using [xk6-redis](https://github.com/grafana/xk6-redis) as an experimental module to keep track of data in Redis:
+
+```javascript
+import { check } from "k6";
+import http from "k6/http";
+import redis from "k6/experimental/redis"; // this will be `k6/x/redis` if you are using it as extension
+import { textSummary } from "https://jslib.k6.io/k6-summary/0.0.1/index.js";
+
+export const options = {
+  scenarios: {
+    usingRedisData: {
+      executor: "shared-iterations",
+      vus: 10,
+      iterations: 200,
+      exec: "measureUsingRedisData",
+    },
+  },
+};
+
+// Instantiate a new redis client
+const redisClient = new redis.Client({
+  addrs: __ENV.REDIS_ADDRS.split(",") || new Array("localhost:6379"), // in the form of "host:port", separated by commas
+  password: __ENV.REDIS_PASSWORD || "",
+});
+
+// Prepare an array of crocodile ids for later use
+// in the context of the measureUsingRedisData function.
+const crocodileIDs = new Array(0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
+
+export function setup() {
+  redisClient.sadd("crocodile_ids", ...crocodileIDs);
+}
+
+export function measureUsingRedisData() {
+  // Pick a random crocodile id from the dedicated redis set,
+  // we have filled in setup().
+  redisClient
+    .srandmember("crocodile_ids")
+    .then((randomID) => {
+      const url = `https://test-api.k6.io/public/crocodiles/${randomID}`;
+      const res = http.get(url);
+
+      check(res, {
+        "status is 200": (r) => r.status === 200,
+        "content-type is application/json": (r) =>
+          r.headers["Content-Type"] === "application/json",
+      });
+
+      return url;
+    })
+    .then((url) => redisClient.hincrby("k6_crocodile_fetched", url, 1));
+}
+
+export function teardown() {
+  redisClient.del("crocodile_ids");
+}
+
+export function handleSummary(data) {
+  redisClient
+    .hgetall("k6_crocodile_fetched")
+    .then((fetched) => Object.assign(data, { k6_crocodile_fetched: fetched }))
+    .then((data) =>
+      redisClient.set(`k6_report_${Date.now()}`, JSON.stringify(data))
+    )
+    .then(() => redisClient.del("k6_crocodile_fetched"));
+
+  return {
+    stdout: textSummary(data, { indent: "  ", enableColors: true }),
+  };
+}
+```
+
+This example also showcases how to write some data and clean up after yourself.
+</details>
+
+TODO maybe have a bunch of issues this can be used to solve 🤔
```

Comment:
```suggestion

TODO maybe have a bunch of issues this can be used to solve 🤔
```

this is in case we decide to not add anything after all - see https://github.com/grafana/k6/pull/2669/files#r964591164